front/temporal/relocation/activities/source_region/core/documents.ts

```
@@ -128,28 +131,59 @@ export async function getDataSourceDocuments({
     },
   };
 
-  // 3) Save the document blobs to file storage.
-  const dataPath = await writeToRelocationStorage(blobs, {
-    workspaceId,
-    type: "core",
-    operation: "data_source_documents_blobs",
-    fileName,
-  });
+  try {
+    // 3) Save the document blobs to file storage.
+    const dataPath = await writeToRelocationStorage(blobs, {
+      workspaceId,
+      type: "core",
+      operation: "data_source_documents_blobs",
+      fileName,
+    });
+
+    localLogger.info(
+      {
+        dataPath,
+        nextPageCursor,
+        nodeCount: nodes.length,
+        totalNodeCount,
+      },
+      "[Core] Retrieved data source documents"
+    );
 
-  localLogger.info(
-    {
+    return {
       dataPath,
       nextPageCursor,
-      nodeCount: nodes.length,
-      totalNodeCount,
-    },
-    "[Core] Retrieved data source documents"
-  );
-
-  return {
-    dataPath,
-    nextPageCursor,
-  };
+      nextLimit: null,
+    };
+  } catch (err) {
+    if (isJSONStringifyRangeError(err)) {
+      const nextLimit = Math.floor(limit / 2);
+      if (nextLimit === 0) {
+        localLogger.error(
+          { error: err, pageCursor, limit, nextLimit },
+          "[Core] Failed to write documents blobs to file storage, string too long - skipping"
+        );
+        // Go to next page, reset limit.
+        return {
```

Wait shouldn't we bail out here? As we don't have any post-relocation tests step, it's better to fail hard than relying on humans to check the logs IMHO.